### PR TITLE
Show privileges as commands

### PR DESCRIPTION
**Pull request by hedengran**
Depends on [neo-technology/neo4j/pull/7515](https://github.com/neo-technology/neo4j/pull/7515)

### CODE REVIEW

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala [hunk 1]
```diff
@@ -167,7 +167,41 @@ class SecurityPrivilegesTest extends DocumentingTest with QueryStatisticsTestSup
         assertStats(r)
       })){}
 
-     }
+      p("Privileges can also be output as Cypher commands with the optional `AS COMMAND[S]`.")
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala]: Having built it and looked at it, this doesn't seem to flow right. I think you maybe want a heading or something to separate the above example from this one. 
Without that, it seems like we're mixing up `SHOW USER` and `SHOW ROLE` in a confusing manner.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala [hunk 2]
```diff
@@ -167,7 +167,41 @@ class SecurityPrivilegesTest extends DocumentingTest with QueryStatisticsTestSup
...
+        Map("command" -> "GRANT NAME MANAGEMENT ON DATABASE * TO `admin`"),
+        Map("command" -> "GRANT START ON DATABASE * TO `admin`"),
+        Map("command" -> "GRANT STOP ON DATABASE * TO `admin`"),
+        Map("command" -> "GRANT TRANSACTION MANAGEMENT (*) ON DATABASE * TO `admin`"),
+        Map("command" -> "GRANT WRITE ON GRAPH * TO `admin`"),
+      ))) {
+        resultTable()
+      }
+
+      p("When showing _user_ privileges as commands, the roles in the Cypher commands will be replace with a parameter.")
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala]: Maybe say why we do this? I know it should be obvious but it's good to be explicit.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala [hunk 3]
```diff
@@ -167,7 +167,41 @@ class SecurityPrivilegesTest extends DocumentingTest with QueryStatisticsTestSup
...
+      query("SHOW USER jake PRIVILEGES AS COMMAND", assertPrivilegeShown(Seq(
+        Map("command" -> "GRANT ACCESS ON DATABASE `neo4j` TO $role"),
+        Map("command" -> "GRANT ACCESS ON DEFAULT DATABASE TO $role"),
+        Map("command" -> "GRANT EXECUTE PROCEDURE * ON DBMS TO $role"),
+      ))) {
+        resultTable()
+      }
+
+      p("It's also possible to have privileges output as revoke commands. " +
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala]: Maybe add something like - 'Like other `SHOW` commands, the output can also be processed using YIELD / WHERE / RETURN'

**Review by mnd999 — APPROVED** [review]
Looks good, just a few suggestions to consider.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala (hunk 1 above)
Comment (hedengran) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala]: Yeah, you're right. I've reorganised the examples now, hopefully you agree that it flows better now.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala [hunk 4]
```diff
@@ -140,6 +140,41 @@ class SecurityPrivilegesTest extends DocumentingTest with QueryStatisticsTestSup
...
+
+      p("Like other `SHOW` commands, the output can also be processed using `YIELD` / `WHERE` / `RETURN`.")
+      query("SHOW ROLE architect PRIVILEGES AS COMMANDS WHERE command CONTAINS 'MATCH'", assertPrivilegeShown(Seq(
+        Map("command" -> "GRANT MATCH {*} ON GRAPH * NODE * TO `architect`"),
+        Map("command" -> "GRANT MATCH {*} ON GRAPH * RELATIONSHIP * TO `architect`"),
+      ))) {
+        resultTable()
+      }
+
+      p("It's also possible to have privileges output as revoke commands. " +
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala]:
```suggestion
      p("It is also possible to have privileges output as revoke commands. " +
```

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @hedengran 👍 

One small editorial change.

**Pull request merged**